src/app-layout/index.tsx

```diff
@@ -269,31 +269,27 @@ const OldAppLayout = React.forwardRef(
       }
     );
 
-    const sizes = (() => {
+    const { items: drawerItems = [] } = drawers || {};
+
+    const getDrawerItemSizes = useCallback(() => {
       const sizes: { [id: string]: number } = {};
-      if (!drawers) {
+      if (!drawerItems) {
         return {};
       }
 
-      for (const item of drawers.items) {
-        if (item.size) {
-          sizes[item.id] = item.size;
+      for (const item of drawerItems) {
+        if (item.initialSize) {
+          sizes[item.id] = sizes[item.id] || item.initialSize;
         }
       }
-
       return sizes;
-    })();
+    }, [drawerItems]);
 
-    const [drawerSizes = {}, setDrawerSizes] = useControllable(
-      Object.keys(sizes).length > 0 ? sizes : undefined,
-      drawers?.onResize,
-      {},
-      {
-        componentName: 'AppLayout',
-        controlledProp: 'drawers.items[].size',
-        changeHandler: 'drawers.onResize',
-      }
-    );
+    const [drawerSizes, setDrawerSizes] = useState(() => getDrawerItemSizes());
+
+    useEffect(() => {
+      setDrawerSizes(prev => ({ ...getDrawerItemSizes(), ...prev }));
+    }, [getDrawerItemSizes]);
```

Comment:
Not sure if this is expected to work this way. When new drawers are being added (for example with the runtime API #1016) it will reset sizes for all previous drawers.

Fine for now, added this comment for future reference

Reply:
I think line 282 was intended to prevent this but with `sizes` being local to that function it will indeed overwrite. I need to re-test this because I did add and remove drawers while testing to confirm that other drawer sizes were left untouched. That said, I likely only added drawers with the default size.

Reply:
Ok, confirmed that this does work as I hoped. the size of drawers is stored in state (the counterpart of `setDrawerSizes`), we are using the callback style of setState so that we can set drawer sizes (to the calculated default sizes, then using any previous values to "override" those values. I could instead filter out any existing from the state update but "apply all and keep existing" felt fine to me. Let me know if you'd like me to change this.